add acitonbar support and use spellid to process check

q513902026 opened the pull request.

q513902026 committed:
Addon ActionBar support

addd addonActionBar support

q513902026 committed:
fix some error

q513902026 committed:
Merge branch 'master' of https://github.com/q513902026/AutoLoh

q513902026 committed:
fix ELVUI ACTIONBAR because OverrideActionBarButton1 is unregisterEvents

Res42 commented:
Thanks for the PR. I really like your ideas but I cannot accept this right now because your modifications broke the addon without NDui or ElvUI.

These are my problems:
- the auto button doesn't work sometimes. Error message is: `You can't do that right now.`
- the auto button doesn't disappear after leaving the vehicle (entering then exiting without completing the quest).

I'll check out your code and see if I can fix these and then I'll merge it.

q513902026 committed:
fix some error and use singleton to nextbutton

1. fix the auto button doesn't disappear after leaving the vehicle (entering then exiting without completing the quest).
2.try fix the auto button doesn't work sometimes. Error message is: You can't do that right now.
I guess the button aren't safe in battle.

q513902026 committed:
Merge branch 'master' of https://github.com/q513902026/AutoLoh

q513902026 commented:
i try fix problems

now nextbutton is only one instance.
this change can fix 

> the auto button doesn't disappear after leaving the vehicle (entering then exiting without completing the quest).

Res42 committed:
Update toc with the new version.

Res42 commented:
I checked out your fix and it fixes the other problem too.

> the auto button doesn't work sometimes. Error message is: `You can't do that right now.`

As I see it, this was the change that fixed it.
`self.nextButton:SetAttribute("type1", "pet");`
`self.nextButton:SetAttribute("action", self:NextStep());`
I think that the previous iteration of this code sometimes wanted to use normal action bar buttons and that triggered the error.

Thank you for your contribution. I'll bump the version and merge your request.
